jwst/coron/median_replace_img.py

```diff
@@ -101,3 +109,34 @@ def median_replace_img(img_model, box_size, bad_bitvalue):
         img_model.data[nimage] = img_int
 
     return img_model
+
+
+def separate_non_science_pixels(img_dq, bad_locations):
+    """
+    For the median filter, we don't care about the NON_SCIENCE pixels, but they
+    produce a ton of warnings that clog up the alignment algorithm and make it
+    run really slowly. In this function, we take all the bad pixels and pull out
+    the ones with NON_SCIENCE flags so we can set them to 0 without running the
+    median filter.
+
+    Parameters
+    ----------
+    img_dq: 2-D dq image
+    bad_locations: 2xN array of flagged pixel indices
+
+    Output
+    ------
+    Tuple of (bad_locations, non_science) pixels
+
+    """
+    def is_science(pix):
+        # return True if pixel is for science, False if flagged NON_SCIENCE
+        val = img_dq[pix[0], pix[1]]
+        flags = dqflags.dqflags_to_mnemonics(val, dqflags.pixel)
+        if "NON_SCIENCE" in flags:
```

Comment:
Alternative method here, which might be faster than using `dqflags_to_mnemonics`, would be something like:
```
flags = np.bitwise_and(val, dqflags.pixel['NON_SCIENCE'])
if flags.any():
    return False
...
```

Reply:
Thanks for the suggestion!